addons/mail/static/src/messaging/entity/composer/composer.js

```diff
@@ -228,20 +319,45 @@ function ComposerFactory({ Entity }) {
         _reset() {
             this.update({
                 attachments: [['unlink-all']],
+                mentionedPartners: [['unlink-all']],
                 textInputContent: '',
                 textInputCursorStart: 0,
                 textInputCursorEnd: 0,
             });
         }
 
+        /**
+         * @override
+         */
+        // _update(data) {
+        //     const {
+        //         activeId = this.activeId || 0,
+        //         textInputContent = this.textInputContent || "",
+        //         textInputCursorStart = this.textInputCursorStart || 0,
+        //         textInputCursorEnd = this.textInputCursorEnd || 0,
+        //     } = data;
+        //
+        //     Object.assign(this, {
+        //         activeId,
+        //         textInputContent,
+        //         textInputCursorStart,
+        //         textInputCursorEnd,
+        //     });
+        // }
+
     }
 
     Composer.entityName = 'Composer';
 
     Composer.fields = {
+        activeId: attr({
+            default: 0,
+        }),
         attachments: many2many('Attachment', {
             inverse: 'composers',
         }),
+        mentionedPartners: one2many('Partner'),
+        mentionSuggestions: one2many('Partner'),
```

Comment:
I would suggest to rename this one to `suggestedPartners`, to better highlight that it does indeed contain partners and not some kind of "suggestion" entity, which would make the rest of the code more obvious at reading.